### PR TITLE
asterisk: update all versions

### PR DESCRIPTION
**Pull request by layus**
Add new Asterisk 16 branch, and update other releases.

/cc maintainers @auntie @DerTim1 @yorickvp

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/asterisk/default.nix [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+  srtp, wget, curl, iksemel, pkgconfig
 }:
 
 let
   common = {version, sha256, externals}: stdenv.mkDerivation rec {
     inherit version;
     name = "asterisk-${version}";
 
-    buildInputs = [ jansson libxml2 libxslt ncurses openssl sqlite utillinux dmidecode libuuid newt lua speex srtp wget curl iksemel ];
+    buildInputs = [ jansson libedit libxml2 libxslt ncurses openssl sqlite utillinux dmidecode libuuid newt lua speex srtp wget curl iksemel pkgconfig ];
```
Comment (Mic92) [inline pkgs/servers/asterisk/default.nix]: `pkgconfig` belongs to `nativeBuildInputs`. It would not hurt to add line breaks in this list.

Comment (Mic92) [reply pkgs/servers/asterisk/default.nix]: `utillinux` is probably also a candidate for `nativeBuildInputs`.

**Review by DerTim1 — APPROVED** [review]

**Comment by DerTim1** [comment]
I think PR #48709  is then superfluous.

**Comment by c0bw3b** [comment]
Weird to end up with `asterisk-lts` > `asterisk-stable` (at least to me)

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by DerTim1** [comment]
This PR can be closed. #55378 is already merged, so this here is superfluous.

**Pull request closed without merging**